src/StateService/MPT/MPTTrie.Find.cs

```diff
@@ -87,23 +95,33 @@ private IEnumerable<(byte[] Key, byte[] Value)> Travers(MPTNode node, byte[] pat
                         var newNode = cache.Resolve(node.Hash);
                         if (newNode is null) throw new InvalidOperationException("Internal error, can't resolve hash when mpt find");
                         node = newNode;
-                        foreach (var item in Travers(node, path))
+                        foreach (var item in Travers(node, path, from))
                             yield return item;
                         break;
                     }
                 case NodeType.BranchNode:
                     {
                         for (int i = 0; i < MPTNode.BranchChildCount; i++)
                         {
-                            foreach (var item in Travers(node.Children[i], i == MPTNode.BranchChildCount - 1 ? path : Concat(path, new byte[] { (byte)i })))
-                                yield return item;
+                            if (i == MPTNode.BranchChildCount - 1 && 0 < from.Length)
```

Comment:
The condition `0 < from.Length` could be outside the `for`

Reply:
Done. Please have a second review.

Reply:
@superboyiii please test it

Reply:
> @superboyiii please test it

Works well.